static/js/hotspots.js

```diff
@@ -2,41 +2,183 @@ var hotspots = (function () {
 
 var exports = {};
 
-exports.show = function (hotspot_list) {
-    $('.hotspot').hide();
-    for (var i = 0; i < hotspot_list.length; i += 1) {
-        $("#hotspot_".concat(hotspot_list[i].name)).show();
-    }
+var HOTSPOT_LOCATIONS = {
+    click_to_reply: {
+        element: '.selected_message .messagebox-content',
+        icon: {
+            top: -71,
+            left: 284,
+        },
+        popover: {
+            top: -210,
+            left: -125,
+            arrow_placement: 'bottom',
+        },
+    },
+    new_topic_button: {
+        element: '#left_bar_compose_stream_button_big',
+        icon: {
+            top: -8,
+            left: -3,
+        },
+        popover: {
+            top: -213,
+            left: -90,
+            arrow_placement: 'bottom',
+        },
+    },
+    stream_settings: {
+        element: '#join_unsub_stream',
+        icon: {
+            top: 7,
+            left: 158.7,
+        },
+        popover: {
+            top: 27,
+            left: -124,
+            arrow_placement: 'top',
+        },
+    },
 };
```

Comment:
If it looks similarly good, I'd definitely go for locations on the element.  Unless I'm mistaken, the next time we tweak our CSS pixels, this will all break, right?